packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -457,4 +474,268 @@ linter:
+      for (String projectPath in directories)
+        printStatus('  ${path.relative(projectPath)}');
+      printStatus('');
+    } else {
+      directories = <String>[Directory.current.path];
+    }
+
+    AnalysisServer server = new AnalysisServer(dartSdkPath, directories);
+    server.onAnalyzing.listen(_handleAnalysisStatus);
+    server.onErrors.listen(_handleAnalysisErrors);
+
+    await server.start();
+
+    int exitCode = await server.onExit;
+    printTrace('Analysis server exiting with code $exitCode.');
+    return 0;
+  }
+
+  bool firstAnalysis = true;
+  Set<String> analyzedPaths = new Set<String>();
+  Map<String, List<AnalysisError>> analysisErrors = <String, List<AnalysisError>>{};
+  Stopwatch analysisTimer;
+  int lastErrorCount = 0;
+
+  void _handleAnalysisStatus(bool isAnalyzing) {
+    if (isAnalyzing) {
+      if (firstAnalysis) {
+        printStatus('Analyzing ${path.basename(Directory.current.path)}...');
+      } else {
+        printStatus('');
+      }
+
+      analyzedPaths.clear();
+      analysisTimer = new Stopwatch()..start();
+    } else {
+      analysisTimer.stop();
+
+      // Sort and print errors.
+      List<AnalysisError> errors = <AnalysisError>[];
+      for (List<AnalysisError> fileErrors in analysisErrors.values)
+        errors.addAll(fileErrors);
+
+      errors.sort();
+
+      for (AnalysisError error in errors)
+        printStatus(error.toString());
+
+      // Print an analysis summary.
+      String errorsMessage;
+
+      int issueCount = errors.length;
+      int issueDiff = issueCount - lastErrorCount;
+      lastErrorCount = issueCount;
+
+      // TODO(devoncarew): If there were no issues found, and no change in the
+      // issue count, do we want to print anything?
+      if (firstAnalysis)
+        errorsMessage = '$issueCount ${pluralize('issue', issueCount)} found';
+      else if (issueDiff > 0)
+        errorsMessage = '$issueDiff new ${pluralize('issue', issueDiff)}, $issueCount total';
+      else if (issueDiff < 0)
+        errorsMessage = '${-issueDiff} ${pluralize('issue', -issueDiff)} fixed, $issueCount remaining';
+      else if (issueCount != 0)
+        errorsMessage = 'no new issues, $issueCount total';
+      else
+        errorsMessage = 'no issues found';
+
+      String files = '${analyzedPaths.length} ${pluralize('file', analyzedPaths.length)}';
+      String seconds = (analysisTimer.elapsedMilliseconds / 1000.0).toStringAsFixed(2);
+      printStatus('$errorsMessage • analyzed $files, $seconds seconds');
+
+      firstAnalysis = false;
+    }
+  }
+
+  void _handleAnalysisErrors(FileAnalysisErrors fileErrors) {
+    fileErrors.errors.removeWhere(_filterError);
+
+    analyzedPaths.add(fileErrors.file);
+    analysisErrors[fileErrors.file] = fileErrors.errors;
+  }
+
+  bool _filterError(AnalysisError error) {
+    // TODO(devoncarew): Also filter the regex items from `analyzeOnce()`.
+
+    if (error.type == 'TODO')
```

Comment:
We're not creating the temporary file with this analysis server path. I'd love to try and move those settings into the _embedded.yaml file.